### ISSUE TITLE
[no-restricted-paths] Add options to ignore type imports

### ISSUE THREAD

**Issue by RebeccaStevens** [benmosher/eslint-plugin-import#1899]
Can an option be added to the `no-restricted-paths` rule to ignore TypeScript's type imports?

```ts
// Allow this.
import type { T } from 'some/restricted/path';

// While still disallowing this.
import { foo } from 'some/restricted/path';
```

My proposal is to add an option at the top level of the config called something along the lines of `allowedImportTypes` which can be set to one of the following options: `"none"` (default) or `"typeImports"`. In future, more different types of imports can be added to this list of options and the option can be set configured to take an array.
Additionally, this option should be available in each "zone" so that individual zones can override this behavior.

I haven't touched this project's code base before but I can probably whip up a PR for this.

**Comment by ljharb** [comment]
Although I'm not a fan of adding typescript-specific options, I think a per-zone option (that's also available higher "up", as well, as you describe) would be good.

I'm not sure if we need an enum - will there ever be another kind of import that's not value or type? - but we can discuss that in a PR.

**Comment by fregante** [comment]
I agree this would be useful. I want to restrict some imports to silo code between contexts; sharing types is obvious since they're used to sync communication between the siloes.

In some cases I can extract them to `./types.ts` and exclude them in batch, but that isn't always practical.

**Comment by golergka** [comment]
I think this is a duplicate of https://github.com/import-js/eslint-plugin-import/issues/2450#issuecomment-1430243181

**Comment by fregante** [comment]
I think so too, but this one is older. I'm ok in keeping the newer one open since it has more thumbs up though

**Comment by golergka** [comment]
My bad, didn't check the date 😅

**Comment by RebeccaStevens** [comment]
I made PR for this years back. It's probably got heaps of conflicts now: #1900

**Comment by OrReuben** [comment]
any update here?
